tests/qchem/of_tests/test_molecular_hamiltonian.py

```diff
@@ -14,7 +14,13 @@
 """
 Unit tests for molecular Hamiltonians.
 """
+import itertools
+
 # pylint: disable=too-many-arguments, protected-access
+import os
+import time
+from multiprocessing import Pool
```

Comment:
Do we need to use `Pool` here? It seems this is better restricted to the internals of the library as a switchable option for parallelism, rather than adding it to the pytest methods.

Reply:
My goal of this is to allow user to use parallelism tool of their choice without being restricted to a single file being written when calculating the Hamiltonian. I agree that putting `Pool` in test is not the best choice, so am keen to hear your idea about where to put it.